Integrations/MicrosoftGraphUser/MicrosoftGraphUser.py

```diff
@@ -0,0 +1,301 @@
+HEADERS = {"Authorization": TOKEN, "Accept": "application/json"}
+USE_SSL = not demisto.params().get('unsecure', False)
+
+''' CONSTANTS '''
+# DEMISTOBOT = "https://demistobot.demisto.com/msg-user" todo: update when ready
+DEMISTOBOT = 'https://ec2-18-197-54-7.eu-central-1.compute.amazonaws.com/msg-user-token/'
+PRODUCT = "MicrosoftGraphUser"
+BLOCK_ACCOUNT_JSON = '{"accountEnabled": false}'
+UNBLOCK_ACCOUNT_JSON = '{"accountEnabled": true}'
+NO_OUTPUTS = {}
+
+
+def camel_case_to_readable(text):
+    """
+    'camelCase' -> 'Camel Case'
+    """
+    if text == 'id':
+        return 'ID'
+    return ''.join(' ' + char if char.isupper() else char.strip() for char in text).strip().title()
+
+
+def parse_outputs(users_data):
+    """
+    Parse user data as received from Microsoft Graph API into Demisto's conventions
+    """
+    if isinstance(users_data, list):
+        users_readable, users_outputs = [], []
+        for user_data in users_data:
+            user_readable = {camel_case_to_readable(k): v for k, v in user_data.items() if k != '@removed'}
+            if '@removed' in user_data:
+                user_readable['Status'] = 'deleted'
+            users_readable.append(user_readable)
+            users_outputs.append({k.replace(' ', ''): v for k, v in user_readable.copy().items()})
+
+        return users_readable, users_outputs
+
+    else:
+        user_readable = {camel_case_to_readable(k): v for k, v in users_data.items() if k != '@removed'}
+        if '@removed' in users_data:
+            user_readable['Status'] = 'deleted'
+        user_outputs = {k.replace(' ', ''): v for k, v in user_readable.copy().items()}
+
+        return user_readable, user_outputs
+
+
+def epoch_seconds(d=None):
+    """
+    Return the number of seconds for given date. If no date, return current.
+    """
+    if not d:
+        d = datetime.utcnow()
+    return int((d - datetime.utcfromtimestamp(0)).total_seconds())
+
+
+def get_token():
+    """
+    Check if we have a valid token and if not get one
+    """
+    ctx = demisto.getIntegrationContext()
+    if ctx.get('token') and ctx.get('stored'):
+        if epoch_seconds() - ctx.get('stored') < 60 * 60 - 30:
+            return ctx.get('token')
+    response = requests.get(
+        DEMISTOBOT,
+        headers=HEADERS,
+        params={"tenant": TENANT, "product": PRODUCT},
+        verify=USE_SSL,
+    )
+    data = response.json()
+    if not response.ok:
+        error_object = json.loads(data.get('detail'))
+        error_details = error_object.get('error_description')
+        if error_details:
+            return_error(error_details)
+        else:
+            return_error(error_object)
+    demisto.setIntegrationContext({'token': data.get('token'), 'stored': epoch_seconds()})
+    return data.get('token')
+
+
+def http_request(method, url_suffix, params=None, body=None):
+    """
+    Generic request to Microsoft Graph
+    """
+    token = get_token()
+    response = requests.request(
```

Comment:
since there is no `try` here, is unreachable error is readable to the user in test and commands? dose it tell the use to check the address?